src/main/java/swerve/ModuleIOReal.java

```diff
@@ -18,87 +22,84 @@ public class ModuleIOReal implements ModuleIO {
-        angleMotor.configMotionAcceleration(motionMagicConfigs[6], Constants.CONFIG_TIMEOUT);
-        angleMotor.configAllowableClosedloopError(0, motionMagicConfigs[7], Constants.CONFIG_TIMEOUT);
-        angleMotor.configMaxIntegralAccumulator(0, motionMagicConfigs[8], Constants.CONFIG_TIMEOUT);
-        angleMotor.configClosedLoopPeakOutput(0, motionMagicConfigs[9], Constants.CONFIG_TIMEOUT);
+        TalonFXConfigurator driveMotorConfigurator = driveMotor.getConfigurator();
+        TalonFXConfigurator angleMotorConfigurator = angleMotor.getConfigurator();
+
+        TalonFXConfiguration driveMotorConfig = new TalonFXConfiguration();
+        driveMotorConfig.Slot0.kP = SwerveConstants.DRIVE_kP;
+        driveMotorConfig.Slot0.kI = SwerveConstants.DRIVE_kI;
+        driveMotorConfig.Slot0.kD = SwerveConstants.DRIVE_kD;
+        driveMotorConfig.Slot0.kV = SwerveConstants.DRIVE_KF;
+        driveMotorConfig.Voltage.PeakForwardVoltage = SwerveConstants.VOLT_COMP_SATURATION;
+        driveMotorConfig.Voltage.PeakReverseVoltage = -SwerveConstants.VOLT_COMP_SATURATION;
+        driveMotorConfig.CurrentLimits.SupplyCurrentLimit = SwerveConstants.SUPPLY_CURRENT_LIMIT.currentLimit;
+        driveMotorConfig.CurrentLimits.StatorCurrentLimit = SwerveConstants.STATOR_CURRENT_LIMIT.currentLimit;
+        driveMotorConfig.Feedback.SensorToMechanismRatio = 1 / SwerveConstants.DRIVE_REDUCTION;
+        driveMotorConfig.Feedback.RotorToSensorRatio = 1;
+        driveMotorConfigurator.apply(driveMotorConfig);
+
+        driveMotor.setNeutralMode(NeutralModeValue.Brake);
+        driveMotor.setInverted(true);
+
+        TalonFXConfiguration angleMotorConfig = new TalonFXConfiguration();
+        angleMotorConfig.Voltage.PeakForwardVoltage = SwerveConstants.VOLT_COMP_SATURATION;
+        angleMotorConfig.Voltage.PeakReverseVoltage = -SwerveConstants.VOLT_COMP_SATURATION;
+        angleMotorConfig.CurrentLimits.SupplyCurrentLimit = SwerveConstants.SUPPLY_CURRENT_LIMIT.currentLimit;
+        angleMotorConfig.CurrentLimits.StatorCurrentLimit = SwerveConstants.STATOR_CURRENT_LIMIT.currentLimit;
+        angleMotorConfig.Feedback.SensorToMechanismRatio = 1 / SwerveConstants.ANGLE_REDUCTION;
+        angleMotorConfig.Feedback.RotorToSensorRatio = 1;
+        angleMotorConfig.MotorOutput.DutyCycleNeutralDeadband = SwerveConstants.NEUTRAL_DEADBAND;
+        angleMotorConfig.Slot0.kP = motionMagicConfigs[0];
+        angleMotorConfig.Slot0.kI = motionMagicConfigs[1];
+        angleMotorConfig.Slot0.kD = motionMagicConfigs[2];
+        angleMotorConfig.Slot0.kV = motionMagicConfigs[3];
+        angleMotorConfig.Slot0.kS = motionMagicConfigs[4];
+        angleMotorConfig.MotionMagic.MotionMagicJerk = motionMagicConfigs[5];
+        angleMotorConfig.MotionMagic.MotionMagicCruiseVelocity = motionMagicConfigs[6];
+        angleMotorConfig.MotionMagic.MotionMagicAcceleration = motionMagicConfigs[7];
```

Comment:
Just change the constructor to accept a `MotionMagicConfigs` object and use it directly like shown [here](https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/closed-loop-requests.html#using-motion-magic-in-api).

Reply:
The MotionMagicConfigs doesn't have slot0 configs in it so it will just be more work. But I will change the PID configs to be like the other motor.

Reply:
True, but you can configure the PID on one line and the motion magic constants on another line.

Reply:
Ok